src/db.rs

```diff
@@ -293,28 +295,43 @@ impl DbInner {
 
     async fn replay_wal(&self) -> Result<(), SlateDBError> {
         let wal_id_last_compacted = self.state.read().state().core.last_compacted_wal_sst_id;
-        let wal_sst_list = self
+        let mut wal_sst_list = self
             .table_store
             .list_wal_ssts((wal_id_last_compacted + 1)..)
             .await?
             .into_iter()
             .map(|wal_sst| wal_sst.id.unwrap_wal_id())
             .collect::<Vec<_>>();
         let mut last_sst_id = wal_id_last_compacted;
+        let sst_batch_size = 4;
+
+        let mut remaining_sst_list = Vec::new();
+        if wal_sst_list.len() > 4 {
+            remaining_sst_list = wal_sst_list.split_off(sst_batch_size);
+        }
+        let mut remaining_sst_iter = remaining_sst_list.iter();
+
+        // Load the first N ssts and instantiate their iterators
+        let mut sst_iterators = VecDeque::new();
         for sst_id in wal_sst_list {
+            sst_iterators.push_back(self.load_sst_iters(sst_id).await?);
+        }
+
+        loop {
```

Comment:
Can this be `while let (mut sst_iter, sst_id) = sst_iterators.pop_front() {`?

Reply:
yes thats much better thanks.